src/zkevm_specs/evm/table.py

```diff
@@ -212,6 +218,27 @@ class CallContextFieldTag(IntEnum):
     StateWriteCounter = auto()
 
 
+class TxLogFieldTag(IntEnum):
+    """
+    Tag for RWTable lookup with tag TxLog, which is used to index specific
+    field of TxLog.
+    """
+
+    # The following are write-only data inside a transaction, they will be written in
+    # State circuit directly.
+    Address = auto()  # address of the contract that generated the event
+    Topics = auto()  # list of topics provided by the contract
+    Data = auto()  # log data in bytes
+    BlockNumber = auto()  # block number which logs are belong to
+    TxHash = auto()  # hash of the transaction
+    TxIndex = auto()  # index of the transaction in the block
```

Comment:
for `BlockNumber` I add comment in `log.py` as following:   
 `omit block number constraint even it is set within op code explicitly, because by default the circuit only handle
    # current block, otherwise, block context lookup is required`. 
Yes,  for `TxIndex ` `TxHash`,  I am planning to remove it as well as `Removed` field since we use `log_index` to identify definite log entry,  then the tx info(tx_hash, tx_id) are also fixed.

Reply:
removed in [0912af8](https://github.com/appliedzkp/zkevm-specs/pull/125/commits/0912af8352fad807c9c70e00c0cd89c1e7532ded)